base/src/main/java/org/aya/normalize/Normalizer.java

```diff
@@ -51,6 +51,8 @@ public final class Normalizer implements UnaryOperator<Term> {
       case StableWHNF _, FreeTerm _ -> postTerm;
       case BetaRedex app -> {
         var result = app.make();
+        // TODO: should we use defaultValue? note that this infect the [MemberCall] case in [TermComparator#doCompareUntyped]
```

Comment:
I think we should use `defaultValue`. When fully normalizing (in which case `defaultValue = postTerm`), we expect everything to be normalized, which includes the arguments.

Reply:
^ I am not sure

Reply:
`someCastTerm` should not unfold to `innerestTerm`

Reply:
If it does then the code is not correct IMO

Reply:
`someCastTerm` doesn't unfold to `innerestTerm` but `someCastTerm .field` unfolds to `innerestTerm .field`. In fact,
`.field` is the following function:
```aya
.field (new impls) = ...
.field (cast inner remember forget) = if field in remember or forget then ... else inner .field
```

Reply:
I see I see I think this reduction is ok.